### PR TITLE
Feature/rep 177 userflows

### PR DESCRIPTION
**Pull request by dovekie**
@alexanderyoung PTAL

### CODE REVIEW

File: bin/critterget.py [hunk 1]
```diff
@@ -53,6 +53,8 @@
 D = 'd'
 ENDPOINTS = 'endpoints'
 DATA = 'data'
+TXN_DURATION = 'P3M'
```
Comment (alexanderyoung) [inline bin/critterget.py]: Why 3 months?

File: bin/critterget.py [hunk 2]
```diff
@@ -520,6 +525,123 @@ def getGenericErrorMon(appId, appName,graph,groupby,messagetype):
...
+    :param app_id: (string) App ID used to request data from the API
+    :param app_name: (string) Human-readable app name
+    :param message_type: (string) Type of message (for sorting in Splunk)
+    :return: None
+    """
+
+    try:
+        for group in userflow_dict['groups']:
+            messages = u''
+            messages += u'(Name="{}",volume={},foregroundTime={}s,failed={},failRate={}%,successful={},revenueAtRisk=${})'.format(
```
Comment (alexanderyoung) [inline bin/critterget.py]: These long lines are really hard to read.  Can we keep these under 80?

File: bin/critterget.py [hunk 3]
```diff
@@ -520,6 +525,123 @@ def getGenericErrorMon(appId, appName,graph,groupby,messagetype):
...
+                                                                  group['name'],
+                                                                  group['series']['startedTransactions']['value'],
```
Comment (alexanderyoung) [inline bin/critterget.py]: I'd add some string literals here so we don't have to worry about typos.

**Comment by alexanderyoung** [comment]
Some formatting comments.

File: bin/critterget.py [hunk 4]
```diff
@@ -598,16 +604,18 @@ def getUserflowsChangeDetails(app_id, app_name, userflow_dict):
...
-                                                                  group['series']['meanForegroundTime']['value'],
-                                                                  group['series']['failedTransactions']['value'],
-                                                                  group['series']['failRate']['value'],
-                                                                  group['series']['succeededTransactions']['value'],
-                                                                  group['series']['failedMoneyValue']['value'])
+            messages += (u'(Name="{}",volume={},foregroundTime={}s,'
+                         u'failed={},failRate={}%,successful={},'
+                         u'revenueAtRisk=${})'.format(
+                              group[NAME],
+                              group[SERIES]['startedTransactions'][VALUE],
```
Comment (alexanderyoung) [inline bin/critterget.py]: Let's get the rest of these into string literals please.

File: bin/critterget.py [hunk 5]
```diff
@@ -628,15 +636,21 @@ def getUserflowsGroups(app_id, app_name, group):
...
+        for transaction in response[SERIES].keys():
             messages = u''
             messages += u'(Metric="{}",count={},rate={}%,moneyValue=${},meanDuration={})'.format(
                 transaction,
-                response['series'][transaction]['count']['value'],
-                response['series'][transaction]['rate']['value'],
-                response['series'][transaction]['moneyValue']['value'],
-                response['series'][transaction]['meanDuration']['value'])
-            print u'{} MessageType={} appName="{}" appId="{}" Userflow="{}" DATA {}'.format(DATETIME_OF_RUN, 'UserflowGroup', app_name, app_id, group, messages)
+                response[SERIES][transaction]['count'][VALUE],
```
Comment (alexanderyoung) [inline bin/critterget.py]: Same comment about string literals above

File: bin/critterget.py [hunk 6]
```diff
@@ -520,6 +531,131 @@ def getGenericErrorMon(appId, appName,graph,groupby,messagetype):
...
+
+
+def getUserflowsDetails(app_id, app_name):
+    """Call the userflows details/change endpoint for an app
+
+    :param app_id: (string) App ID used to request data from the API
+    :param app_name: (string) Human-readable app name
+    :return: None
+    """
+    uri = 'transactions/{}/details/change/P1M?pageNum=1&pageSize=10&sortBy=name&sortOrder=ascending'.format(app_id)
```
Comment (alexanderyoung) [inline bin/critterget.py]: Looks like we're getting the top 10 in the last month, is that what we want?

Comment (dovekie) [inline bin/critterget.py]: This is the query that is used to populate the equivalent table on the portal. I want to make sure the data match as much as possible.

**Comment by alexanderyoung** [comment]
LGTM thanks

**Pull request merged**